### PR TITLE
Connect User Endpoints

### PR DESCRIPTION
**Pull request by Sparye**
# Description
connect user endpoints including user login, sign up, forget password, logout

## Related Issue

Solves #
#84 Connect to Backend 

## Type of change

- [x] New feature (enhancement)
- [ ] Bug fix
- [ ] Refactor
- [ ] Breaking change
- [ ] Documentation

## How Has This Been Tested?

- [ ] Automated testing
- [ ] Manual testing

# Checklist:

- [ ] Does a similar (open or closed) pull request not already exist?
- [ ] Is the pull request _head_ repository a fork repository?
- [ ] Is the pull request _compare_ branch a development branch?
- [ ] Is the code documented, particularly in hard-to-understand areas?
- [ ] Does the code build without new warnings?
- [ ] Has testing been performed that proves changes are effective and work?
- [ ] Has a self- and/or peer-review of the code been performed?
- [ ] Have dependent changes been merged and published in downstream modules?
- [ ] Does all new and existing automated testing pass?
- [ ] Is the person responsible for the repository assigned to the pull request?
- [ ] Is the pull request linked to a project?
- [ ] Is the pull request linked to a milestone?

For more information, refer to the Contributing Guidelines and Code of Conduct links at the bottom of this page.

### CODE REVIEW

File: src/pages/LoginRegister/RegisterPage.jsx [hunk 1]
```diff
@@ -3,30 +3,44 @@ import "./LoginPage.css";
...
+              {
+                email: details.email,
+                username: details.username,
+                password: details.password1,
+              },
+            ]);
+            alert("Registered successfully! Please Log in!");
+          }
+        } catch (error) {
+          console.log("failed to sign up");
```
Comment (GlockTian) [inline src/pages/LoginRegister/RegisterPage.jsx]: as alert?

File: src/pages/LoginRegister/ForgotPasswordPage.jsx [hunk 2]
```diff
@@ -1,9 +1,26 @@
...
+
+  const createRestPassword = useMutation('/api/v1/users/', {
+    method: 'patch',
+    pathParams: { id: id ?? '' },
+  })
+
+  async function handleResetPassword(email) {
+    await createRestPassword({
+      body: {
+        email: 'admin@123',
```
Comment (GlockTian) [inline src/pages/LoginRegister/ForgotPasswordPage.jsx]: probably should be email? I mean use the variable

File: src/pages/LoginRegister/RegisterPage.jsx (hunk 1 above)
Comment (Sparye) [reply src/pages/LoginRegister/RegisterPage.jsx]: I will change it to toast

**Comment by frasermcc9** [comment]
Please add a description

File: src/pages/LoginRegister/LoginPage.jsx [hunk 3]
```diff
@@ -4,26 +4,31 @@ import { useNavigate } from "react-router-dom";
...
+    const loginDetails = {
+      data: {
+        email: details.email,
+        plaintextPassword: details.password,
+      },
+    };
+    try {
+      const response = await userLogin(loginDetails);
+      console.log(response);
+      localStorage.setItem("id", response.userID);
```
Comment (frasermcc9) [inline src/pages/LoginRegister/LoginPage.jsx]: IMO this should be saved to localstorage as a single object (json stringified)

File: src/pages/LoginRegister/LoginPage.jsx [hunk 4]
```diff
@@ -4,26 +4,31 @@ import { useNavigate } from "react-router-dom";
...
+  const LoginFormDetail = async (details) => {
+    const loginDetails = {
+      data: {
+        email: details.email,
+        plaintextPassword: details.password,
+      },
+    };
+    try {
+      const response = await userLogin(loginDetails);
+      console.log(response);
```
Comment (frasermcc9) [inline src/pages/LoginRegister/LoginPage.jsx]:
```suggestion
```

File: src/pages/LoginRegister/ForgotPasswordPage.jsx [hunk 5]
```diff
@@ -27,7 +45,10 @@ export default function ForgotPasswordPage() {
             className="LoginRegisterForm_button"
             id="sub_btn"
             type="submit"
-          >
+            onClick = { () => {
+              const email = document.getElementsByClassName('userEmail').value
```
Comment (frasermcc9) [inline src/pages/LoginRegister/ForgotPasswordPage.jsx]: Avoid imperative code like this.

Either useRef to get the element, or store the the value in state.

File: src/pages/LoginRegister/LoginPage.jsx [hunk 6]
```diff
@@ -4,26 +4,31 @@ import { useNavigate } from "react-router-dom";
...
-    } else {
-      console.log("No such account, Please register first!");
-      alert("No such account, Please register first!");
+  const LoginFormDetail = async (details) => {
+    const loginDetails = {
+      data: {
+        email: details.email,
+        plaintextPassword: details.password,
+      },
+    };
```
Comment (frasermcc9) [inline src/pages/LoginRegister/LoginPage.jsx]:
```suggestion
  const LoginFormDetail = async ({email, password: plaintextPassword}) => {
    const loginDetails = {
      data: {
        email,
        plaintextPassword,
      },
    };
```

Check this works first :P

**Review by frasermcc9 — CHANGES_REQUESTED** [review]

**Review by frasermcc9 — DISMISSED** [review]

**Review by frasermcc9 — CHANGES_REQUESTED** [review]
Can you make the register page go back to the login page if the register was successful?

**Review by frasermcc9 — APPROVED** [review]

**Pull request merged**